pkg/destroy/openstack/openstack.go

```
@@ -449,31 +457,108 @@ func updateFips(allFIPs []floatingips.FloatingIP, opts *clientconfig.ClientOpts,
+
 	if err != nil {
 		logger.Error(err)
-		return false, nil
+		return err
+	}
+
+	fips, err := floatingips.ExtractFloatingIPs(fipPages)
+	if err != nil {
+		logger.Error(err)
+		return err
+	}
+
+	for _, fip := range fips {
+		logger.Debugf("Deleting FIP %q", fip.ID)
+		err = floatingips.Delete(conn, fip.ID).ExtractErr()
+		if err != nil {
+			// Ignore the error if the FIP cannot be found
+			var gerr gophercloud.ErrDefault404
+			if !errors.As(err, &gerr) {
+				logger.Errorf("Deleting FIP %q failed: %v", fip.ID, err)
+				return err
+			}
+			logger.Debugf("Cannot find FIP %q. It's probably already been deleted.", fip.ID)
+		}
+	}
+	return nil
+}
+
+func getRouters(opts *clientconfig.ClientOpts, filter Filter, logger logrus.FieldLogger) ([]routers.Router, error) {
+	conn, err := clientconfig.NewServiceClient("network", opts)
+	if err != nil {
+		return nil, err
 	}
 	tags := filterTags(filter)
 	listOpts := routers.ListOpts{
 		TagsAny: strings.Join(tags, ","),
 	}
 
 	allPages, err := routers.List(conn, listOpts).AllPages()
+	if err != nil {
+		return nil, err
+	}
+
+	allRouters, err := routers.ExtractRouters(allPages)
+	if err != nil {
+		return nil, err
+	}
+	return allRouters, nil
+}
+
+func clearRoutersInferfaces(opts *clientconfig.ClientOpts, filter Filter, logger logrus.FieldLogger) (bool, error) {
+	logger.Debug("Clearing openstack routers interfaces")
+	defer logger.Debugf("Exiting clearing openstack routers interfaces")
+
+	conn, err := clientconfig.NewServiceClient("network", opts)
+	if err != nil {
+		return false, err
+	}
+
+	allRouters, err := getRouters(opts, filter, logger)
 	if err != nil {
 		logger.Error(err)
 		return false, nil
 	}
 
-	allRouters, err := routers.ExtractRouters(allPages)
+	numberToClear := len(allRouters)
+	numberCleaned := 0
+	for _, router := range allRouters {
+		_, err = removeRouterInterfaces(conn, filter, router, logger)
+		if err != nil {
+			continue
+		}
```

Do you want to `logger.Debug` the error instead of silently ignoring?

---

It was like this before [1], but yeah.

[1] https://github.com/openshift/installer/pull/5050/files/d87b0c6138b17303dd4e3e195cf15d7e735e8629#diff-e480f6c0393b6788fe8352f4f8bdc63f021f9e84de632a21607aab8b414c6e3fL512-L515